### DISCUSSION TITLE
SA_Score gives different values depending if the molecule has explicit hydrogens

### DISCUSSION THREAD

**Discussion by ale94mleon** [rdkit/rdkit#7047] [Q&A] [answered]
I was not expecting this behaviour, because I assume that the ease of synthesis does not depend if the molecule has the Hs or not (in the reality it always does 😅).  However I am not sure what characteristics should have the molecule to be compatible with the estimation of the score, should the molecule have the explicit hydrogen or not during the calculation?

Here the code to reproduce.

```python
def import_sascorer():
    # In order to import sascorer from RDConfig.RDContribDir
    import importlib.util as importlib_util

    from rdkit.Chem import RDConfig
    import os
    spec = importlib_util.spec_from_file_location(
        'sascorer', os.path.join(RDConfig.RDContribDir, 'SA_Score', 'sascorer.py'))
    sascorer = importlib_util.module_from_spec(spec)
    spec.loader.exec_module(sascorer)
    return sascorer

sascorer = import_sascorer()

mol = Chem.MolFromSmiles('CCCCCCC(O)CNCC')
mol_h = Chem.AddHs(mol)

print(sascorer.calculateScore(mol), sascorer.calculateScore(mol_h))
```
Result:

```
2.478963513446077 8.14053388631115
```

**Comment by greglandrum** [comment, marked as answer]
The SA score code, like most of the rest of the RDKit code, is designed to be called on molecules that do not have explicit Hs.
